packages/app/src/app/overmind/namespaces/editor/actions.ts

```diff
@@ -41,6 +44,57 @@ export const internal = internalActions;
 
 export const onNavigateAway: Action = () => {};
 
+export const persistCursorToUrl: Action<{
+  module: Module;
+  selection?: UserSelection;
+}> = debounce(({ effects }, { module, selection }) => {
+  effects.router.setParameter('file', module.path);
+  if (selection?.primary?.selection?.length) {
+    const [head, anchor] = selection.primary.selection;
+    const serializedSelection = head + ':' + anchor;
+    effects.router.setParameter('selection', serializedSelection);
+  } else {
+    effects.router.setParameter('selection', null);
+  }
+}, 500);
+
+export const loadCursorFromUrl: AsyncAction = async ({
+  effects,
+  actions,
+  state,
+}) => {
+  if (!state.editor.currentSandbox) {
+    return;
+  }
+  try {
```

Comment:
Any reason for the `try/catch`? 🤔

Reply:
This is user input, anything could go wrong here that we haven't foreseen, and if this breaks it would crash the whole editor. I do think I should've put this into a `try...catch` in the call instead of implementation, will change that...